nixos/modules/services/networking/sickbeard.nix

```diff
@@ -0,0 +1,79 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.sickbeard;
+  sickbeard = cfg.package;
+in
+{
+
+  ###### interface
+
+  options = {
+    services.sickbeard = {
+      enable = mkOption {
+        default = false;
+        description = "Whether to enable the sickbeard server.";
+      };
+      package = mkOption {
+        type = types.package;
+        default = pkgs.sickbeard;
+        example = literalExample "pkgs.sickrage";
+        description =''
+          Enable <literal>pkgs.sickrage</literal> or <literal>pkgs.sickgear</literal>
+          as an alternative to SickBeard
+        '';
+      };
+      dataDir = mkOption {
+        default = "/var/lib/sickbeard";
+        description = "Path where to store data files.";
+      };
+      configFile = mkOption {
+        default = "/var/lib/sickbeard/config.ini";
+        description = "Path to config file.";
+      };
+      port = mkOption {
+        default = 8081;
+        description = "Port to bind to.";
+      };
+      user = mkOption {
+        default = "sickbeard";
+        description = "User to run the service as";
+      };
+      group = mkOption {
+        default = "sickbeard";
+        description = "Group to run the service as";
+      };
+    };
+  };
+
+
+  ###### implementation
+
+  config = mkIf cfg.enable {
+
+    users.users.sickbeard = {
```

Comment:
In the options you allow the user to specify username but don't act on it. If I wanted to run sickgear as my own user I could not. See the user and group creation from here as an example:

https://github.com/NixOS/nixpkgs/blob/980cbff93c20b5dcd4ac63384db84490f5e07a45/nixos/modules/services/misc/redmine.nix